serverlessworkflow/modules/ROOT/pages/getting-started/cncf-serverless-workflow-specification-support.adoc

```diff
@@ -1 +1,368 @@
+----
+
+Later in the `state` definition, you can call it this way:
+
+.Example of a function reference within a state
+[source,json]
+----
+{
+  "states": [
+    {
+      "name": "myState",
+      "type": "operation",
+      "actions": [
+        {
+          "name": "printAction",
+          "functionRef": {
+            "refName": "printMessage",
+            "arguments": {
+              "message": "."
+            }
+          }
+        }
+      ]
+    }
+  ]
+}
+----
+
+You should see the data output in the console.
+
+[#func_java]
+==== Java Custom Function
+
+Kogito supports calling Java functions within the Maven project, in which you define the workflow. For example, you can declare your functions like this:
+
+.Example of a Java function declaration
+[source,json]
+----
+{
+  "functions": [
+    {
+      "name": "myFunction", <1>
+      "type": "custom", <2>
+      "operation": "service:java:com.acme.MyInterfaceOrClass::myMethod" <3>
+    }
+  ]
+}
+----
+
+1. The function name
+2. The `custom` function type
+3. The Kogito custom operation definition. `service` is the reserved operation keyword followed by the other keyword `java`. The next portion is the FQDN of the interface or implementation class, followed by the method name.
+
+Your method's interface *must* receive a https://github.com/FasterXML/jackson[Jackson's] `JsonNode` object and return either `void` or another `JsonNode`. For example:
+
+.Example of a Java function implementation
+[source,java]
+----
+public class MyInterfaceOrClass {
+
+    public void myMethod(JsonNode workflowData) {
+        // do whatever I want with the JsonNode:
+        // { "workflowdata": {} }
+    }
+    
+    public JsonNode myMethod(JsonNode workflowData) {
+        // do whatever I want with the JsonNode:
+        // { "workflowdata": {} }
+        // return the modified content:
+        return workflowData;
+    }
+}
+----
+
+To call this function within your workflow, you can extract the JSON value you need via a xref:core/understanding-jq-expressions.adoc[`jq` expression] or pass it without any arguments. In this case, the caller sends the whole payload. For example:
+
+.Example of a reference to a Java function in the Operation state
+[source,json]
+----
+{
+  "states": [
+    {
+      "name": "myState",
+      "type": "operation",
+      "actions": [
+        {
+          "name": "callJavaFunctionAction",
+          "functionRef": {
+            "refName": "myFunction"
+          }
+        }
+      ]
+    }
+  ]
+}
+----
+
+Alternatively, you can only pass the necessary data:
+
+.Example of a reference to a Java function in the Operation state passing manipulated data
+[source,json]
+----
+{
+  "states": [
+    {
+      "name": "myState",
+      "type": "operation",
+      "actions": [
+        {
+          "name": "callJavaFunctionAction",
+          "functionRef": {
+            "refName": "myFunction",
+            "arguments": {
+              "data": "${ .my.expression.to.data }"
```

Comment:
maybe we can provide a working example for this use case?

Reply:
I'll once I break this part into a new guide as requested by @ibek

Reply:
The example is already described in the guide.